website/src/documentation/runtime/environments.md

```diff
@@ -0,0 +1,187 @@
+
+## Building container images
+
+Before building Beam SDK container images:
+* Register a [Bintray](https://bintray.com/) account with a Docker repository named `apache`.
+* Install [Docker](https://www.docker.com/) on your workstation.
+
+To build Beam SDK container images:
+
+<ol>
+    <li>
+        Navigate to your local copy of the <a href="https://github.com/apache/beam"><code>beam</code></a>
+    </li>
+    <li>
+        Run Gradle with the <code>docker</code> target: <pre>./gradlew docker</pre>
+    </li>
+</ol>
+
+> **Note**: It may take a long time to build all of the container images. You can instead build the images for specific SDKs:
+>
+> ```
+> ./gradlew -p sdks/java/container docker
+> ./gradlew -p sdks/python/container docker
+> ./gradlew -p sdks/go/container docker
+> ```
+
+Run `docker images` to examine the containers. For example, if you successfully built the container images, the command prompt displays a response like:
+
+```
+REPOSITORY                                       TAG                    IMAGE ID            CREATED       SIZE
+$USER-docker-apache.bintray.io/beam/python     latest             4ea515403a1a      3 minutes ago     1.27GB
+$USER-docker-apache.bintray.io/beam/java       latest             0103512f1d8f     34 minutes ago      780MB
+$USER-docker-apache.bintray.io/beam/go         latest             ce055985808a     35 minutes ago      121MB
+```
+
+<b>Although the respository names look like URLs, the container images are stored locally on your workstation.</b> After building the container images locally, you can [push](#pushing-container-images) them to an eponymous repository online.
+
+### Overriding default Docker targets
+
+The default SDK version is `latest` and the default Docker repository is the following Bintray location:
+
+```
+$USER-docker-apache.bintray.io/beam
+```
+
+When you [build SDK container images](#building-container-images), you can override the default version and location.
+
+To specify an older Python SDK version, like 2.3.0, build the container with the `docker-tag` option:
+
+```
+./gradlew docker -Pdocker-tag=2.3.0
+```
+
+To change the `docker` target, build the container with the `docker-repository-root` option:
+
+```
+./gradlew docker -Pdocker-repository-root=$LOCATION
+```
+
+## Customizing container images
+
+You can add extra dependencies or serialization files to container images so the execution engine doesn't need them.
+
+To customize a container image, either:
+* [Write a new](#writing-new-dockerfiles) [Dockerfile](https://docs.docker.com/engine/reference/builder/) on top of the original
+* [Modify](#modifying-dockerfiles) the [original Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile) and reimage the container
+
+It's often easier to write a new Dockerfile, but you can customize anything, including the base OS, by modifying the original.
```

Comment:
"It's often easier to write a new Dockerfile, but by modifying the original Dockerfile, you can customize anything (including the base OS) ."

Just a suggestion--I don't really like parentheses, but this sentence is confusing with too many clauses. Another possibility is to break it up.

"It's often easier to write a new Dockerfile. However, by modifying the original Dockerfile, you can customize anything, including the base OS."

Reply:
Done. Split it into two sentences - much more digestible